src/System.Private.ServiceModel/tests/Scenarios/SelfHostWcfService/Program.cs

```diff
@@ -12,14 +12,32 @@ private static void Main(string[] args)
         {
             using (ServiceHost wcfservicehost = new ServiceHost(typeof(WcfService)))
             using (ServiceHost userNameService = new ServiceHost(typeof(WcfUserNameService)))
+            using (ServiceHost wcfDuplexService = new ServiceHost(typeof(WcfDuplexService)))
             {
                 wcfservicehost.Open();
                 Console.WriteLine("WcfService is ready");
                 userNameService.Open();
                 Console.WriteLine("Wcf user name service is ready");
+                wcfDuplexService.Open();
+                Console.WriteLine("WcfDuplexService is ready");
+
+                DumpEndpoints(wcfservicehost, userNameService, wcfDuplexService);
+
                 Console.WriteLine("Press ENTER to close services...");
                 Console.ReadLine();
             }
         }
+
+         static void DumpEndpoints(params ServiceHost[] hosts)
+         {
```

Comment:
:+1: on `DumpEndpoints`. 

Not important: Is there a way that `DumpEndpoints` can just enumerate through all ServiceHosts present on the system (or app domain) rather than have to pass this information in?